### PR TITLE
Integration between current algorithms

### PR DESCRIPTION
**Pull request by PaulaViriato**
Addition of backtranslation algorithm, addition of deanonymization algorithm, integration between all algorithms, generation of a general dialog augmentation using all algorithms currently in the project.

### CODE REVIEW

File: backtranslation.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        elif (tradutor == 6):
+            trad = ts.translate_html(original, translator=ts.sogou, from_language='pt', to_language=idioma, n_jobs=1, timeout=20.0)
+            result = ts.translate_html(trad, translator=ts.sogou, from_language=idioma, to_language='pt', n_jobs=1, timeout=20.0)
+        elif (tradutor == 7):
+            trad = ts.translate_html(original, translator=ts.deepl, from_language='pt', to_language=idioma, n_jobs=1, timeout=20.0)
+            result = ts.translate_html(trad, translator=ts.deepl, from_language=idioma, to_language='pt', n_jobs=1, timeout=20.0)
+    except: result = ""
+    return result
+
+def parallel_translation (original, rate, idioma, tradutor, currtags, traducoes):
```
Comment (jadermcs) [inline backtranslation.py]: In function declaration the Python style guide says to not put spaces between name and parenthesis, it should be in this case parallel_translation(...).

File: util.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    return result
+
+def organizeturn (turn, delex):
+    new_turn = turn.copy()
+    new_turn["utterance"] = delex
+    new_turn["utterance_delex"] = delex
+    new_turn["slot-values"] = turn["slot-values"].copy()
+    return new_turn
+
+def strTimeProp (start, end, format, prop):
```
Comment (jadermcs) [inline util.py]: python uses snake case, not camelcase, so the name should be str_time_prop(...)

File: mada.py [hunk 3]
```diff
@@ -1,79 +1,93 @@
...
+import random
+import json
+import util
 random.seed(20211109)
 
-def augment(sentence):
-    return eda(sentence, ["[cpf]", "[placa]", "[valor]"])
-
-def bfs(pflow, intent_sample):
+def dfs (pflow, intent_sample, size, example, rate = 0.93, silence = False, mode = "test"):
```
Comment (jadermcs) [inline mada.py]: This algorithm is now a Breadth-First Search, not a Depth anymore.

**Review by jadermcs — CHANGES_REQUESTED** [review]
The code should improve style before merged.

File: backtranslation.py (hunk 1 above)
Comment (PaulaViriato) [reply backtranslation.py]: Solved.

File: util.py (hunk 2 above)
Comment (PaulaViriato) [reply util.py]: Solved.

File: mada.py (hunk 3 above)
Comment (PaulaViriato) [reply mada.py]: Solved.

**Review by jadermcs — APPROVED** [review]

**Pull request merged**